src/cmd/run.js

```diff
@@ -107,33 +108,33 @@ export function defaultFirefoxClient(
     maxRetries=250, retryInterval=120,
   }: CreateFirefoxClientParams = {}
 ): Promise<RemoteFirefox> {
-  var retries = 0;
-
-  function establishConnection() {
-    return new Promise((resolve, reject) => {
-      connectToFirefox()
-        .then((connectedClient) => {
-          log.debug('Connected to the Firefox debugger');
-          resolve(connectedClient);
-        })
-        .catch(onlyErrorsWithCode('ECONNREFUSED', (error) => {
-          if (retries >= maxRetries) {
-            log.debug('Connect to Firefox debugger: too many retries');
-            throw error;
-          } else {
-            setTimeout(() => {
-              retries ++;
-              log.debug(
-                `Retrying Firefox (${retries}); connection error: ${error}`);
-              resolve(establishConnection());
-            }, retryInterval);
-          }
-        }))
-        .catch((error) => {
+  async function establishConnection() {
+    var retries = 0;
+    var lastError;
+
+    while (retries <= maxRetries) {
```

Comment:
bah! Nevermind. That doesn't give you a counter. Looks like there isn't a one-liner equivalent of Python's `range()`.